cyclonedx/model/component.py

```diff
@@ -60,6 +232,486 @@ class ComponentType(Enum):
+    modified, redistributed, combined with other components, etc. Pedigree supports viewing this complex chain from the
+    beginning, the end, or anywhere in the middle. It also provides a way to document variants where the exact relation
+    may not be known.
+
+    .. note::
+        See the CycloneDX Schema definition: https://cyclonedx.org/docs/1.4/xml/#type_pedigreeType
+    """
+
+    def __init__(self, ancestors: Optional[List['Component']] = None, descendants: Optional[List['Component']] = None,
+                 variants: Optional[List['Component']] = None, commits: Optional[List[Commit]] = None,
+                 patches: Optional[List[Patch]] = None, notes: Optional[str] = None) -> None:
+        if not ancestors and not descendants and not variants and not commits and not patches and not notes:
+            raise NoPropertiesProvidedException(
+                'At least one of `ancestors`, `descendants`, `variants`, `commits`, `patches` or `notes` must be '
+                'provided for `Pedigree`'
+            )
+
+        self.ancestors = ancestors
+        self.descendants = descendants
+        self.variants = variants
+        self.commits = commits
+        self.patches = patches
+        self.notes = notes
+
+    @property
+    def ancestors(self) -> Optional[List['Component']]:
+        """
+        Describes zero or more components in which a component is derived from. This is commonly used to describe forks
+        from existing projects where the forked version contains a ancestor node containing the original component it
+        was forked from.
+
+        For example, Component A is the original component. Component B is the component being used and documented in
+        the BOM. However, Component B contains a pedigree node with a single ancestor documenting Component A - the
+        original component from which Component B is derived from.
+
+        Returns:
+            List of `Component` if set else `None`
+        """
+        return self._ancestors
+
+    @ancestors.setter
+    def ancestors(self, ancestors: Optional[List['Component']]) -> None:
+        self._ancestors = ancestors
+
+    def add_ancestor(self, ancestor: 'Component') -> None:
+        """
+        Adds an ancestor.
+
+        Args:
+            ancestor:
+                `Component`
+        """
+        self.ancestors = (self.ancestors or []) + [ancestor]
+
+    @property
+    def descendants(self) -> Optional[List['Component']]:
+        """
+        Descendants are the exact opposite of ancestors. This provides a way to document all forks (and their forks) of
+        an original or root component.
+
+        Returns:
+            List of `Component` if set else `None`
+        """
+        return self._descendants
+
+    @descendants.setter
+    def descendants(self, descendants: Optional[List['Component']]) -> None:
+        self._descendants = descendants
+
+    def add_descendant(self, descendant: 'Component') -> None:
```

Comment:
again: if the internal property was a list at all times, 
this mehod would not be needed, because people could call `object.descendants.append(foo)` to modify internal ata structures.
lass code, mor clear data structires.